frontend/packages/integration-tests-cypress/tests/monitoring/monitoring.spec.ts

```diff
@@ -0,0 +1,153 @@
+  });
+
+  afterEach(() => {
+    checkErrors();
+  });
+
+  after(() => {
+    cy.deleteProject(testName);
+    cy.logout();
+  });
+
+  // each it() should be its own atomic test (run independently of other tests), per best practices.
+  it('displays and filters the Alerts list page, links to detail pages', () => {
+    cy.log('use vert. nav. menu to goto Monitoring -> Alerting');
+    cy.clickNavLink(['Monitoring', 'Alerting']);
+    // TODO, switch to 'listPage.titleShouldBe('Alerting');', when we switch to new test id
+    cy.byLegacyTestID('resource-title').should('have.text', 'Alerting');
+    listPage.projectDropdownShouldNotExist();
+
+    cy.log('filter Alerts');
+    listPage.filter.byName('Watchdog');
+    listPage.rows.countShouldBe(1);
+
+    cy.log('drills down to Alert details page');
+    listPage.rows.shouldExist('Watchdog').click();
+    shouldBeWatchdogAlertDetailsPage();
+
+    cy.log('drill down to the Alerting Rule details page');
+    cy.byTestID('alert-rules-detail-resource-link')
+      .contains('Watchdog')
+      .click();
+    shouldBeWatchdogAlertRulesPage();
+
+    cy.log('drill back up to the Alert details page');
+    // Active Alerts list should contain a link back to the Alert details page
+    cy.byTestID('active-alerts')
+      .first()
+      .click();
+    shouldBeWatchdogAlertDetailsPage();
+  });
+
+  it('creates and expires a Silence', () => {
+    // each it() should be its own atomic test, likely start with cy.visit(the-page) or nav to page
+    cy.visit('monitoring/alerts');
+    cy.log('filter to Watchdog alert');
+    listPage.filter.byName('Watchdog');
+    listPage.rows.countShouldBe(1);
+    listPage.rows.shouldExist('Watchdog').click();
+    shouldBeWatchdogAlertDetailsPage();
+
+    cy.log('silence Watchdog alert');
+    // After creating the Silence, should be redirected to its details page
+    detailsPage.clickPageActionButton('Silence Alert');
+    // launches page form
+    cy.byTestID('start-immediately').should('be.checked');
+    cy.byTestID('from').should('have.value', 'Now');
+    cy.byLegacyTestID('dropdown-button').should('contain', '2h');
+    cy.byTestID('until').should('have.value', '2h from now');
+    // Change duration
+    cy.byLegacyTestID('dropdown-button')
+      .click()
+      .get(`[data-test-dropdown-menu="1h"]`)
+      .click();
+    cy.byTestID('until').should('have.value', '1h from now');
+    // Change to not start now
+    cy.byTestID('start-immediately').click();
+    cy.byTestID('start-immediately').should('not.be.checked');
+    // Allow for some difference in times
+    cy.byTestID('from').should('not.have.value', 'Now');
+    // eslint-disable-next-line promise/catch-or-return
+    cy.byTestID('from').then(($fromElement) => {
+      const fromText = $fromElement[0].getAttribute('value');
+      expect(Date.parse(fromText) - Date.now()).toBeLessThan(10000);
+      // eslint-disable-next-line promise/catch-or-return,promise/no-nesting
```

Comment:
We need to fix these lint errors instead of disabling them.

Reply:
I was able to remove disabling of `promise/catch-or-return`, by doing `return cy....` in places (can't do a catch() as Cypress commands aren't Promises -see below and [here](https://github.com/cypress-io/cypress/issues/1417#issuecomment-370860080) ).  

There are still places I need to nest `.then()`'s